mitiq/folding_mitiq.py

```diff
@@ -0,0 +1,82 @@
+import random
+
+
+def deduce_platform(circuit):
```

Comment:
This introduces some implicit dependencies on how Cirq and pyQuil manipulate programs. Without thinking this through a lot further I'd be hesitant to program in this way. Would recommend leaving this out of the commit for now as it opens up a lot of other questions. Would be happy to start a new issue/discussion if you think that this kind of abstraction is the way we should go @andreamari